src/islpr.cc

```diff
@@ -0,0 +1,401 @@
+  isl_map* imapR = NULL;
+  isl_map* imapK = NULL;
+  imapR = isl_map_read_from_str(ctx, relstr.c_str());
+  imapK = isl_map_read_from_str(ctx, knownstr.c_str());
+
+  imapR = isl_map_union( imapR, imapK);
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_map(ip ,imapR);
+  cstr=isl_printer_get_str(ip);
+  string stringFromISL = cstr;
+/*  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_map_free(imapR);
+  imapR= NULL;
+  isl_map_free(imapK);
+  imapK= NULL;
+  isl_ctx_free(ctx); 
+*/
+  return stringFromISL;
+
+}
+
+string getRelationStringFromISL(string relstr)
+{
+  // Get an isl context
+  isl_ctx *ctx;
+  ctx = isl_ctx_alloc();
+  
+  // Get an isl printer and associate to an isl context
+  isl_printer * ip = NULL;
+  ip = isl_printer_to_str(ctx);
+  
+  // load Relation r into ISL map
+  isl_map* imap = NULL;
+  imap = isl_map_read_from_str(ctx, relstr.c_str());
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_map(ip ,imap);
+  cstr=isl_printer_get_str(ip);
+  string stringFromISL = cstr;
+  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_map_free(imap);
+  imap= NULL;
+  isl_ctx_free(ctx); 
+
+  return stringFromISL;
+}
+string getSetStringFromISL(string relstr)
+{
+  // Get an isl context
+  isl_ctx *ctx;
+  ctx = isl_ctx_alloc();
+  
+  // Get an isl printer and associate to an isl context
+  isl_printer * ip = NULL;
+  ip = isl_printer_to_str(ctx);
+  
+  // load Relation r into ISL map
+  isl_map* imap = NULL;
+  imap = isl_map_read_from_str(ctx, relstr.c_str());
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_map(ip ,imap);
+  cstr=isl_printer_get_str(ip);
+  string stringFromISL = cstr;
+  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_map_free(imap);
+  imap= NULL;
+  isl_ctx_free(ctx); 
+
+  return stringFromISL;
+}
+
+string project_out (string relstr, unsigned pos)
+{
+  // Get an isl context
+  isl_ctx *ctx;
+  ctx = isl_ctx_alloc();
+  
+  // Get an isl printer and associate to an isl context
+  isl_printer * ip = NULL;
+  ip = isl_printer_to_str(ctx);
+  
+  // load Relation r into ISL map
+  isl_map* imap = NULL;
+  imap = isl_map_read_from_str(ctx, relstr.c_str());
+
+  imap = isl_map_project_out(imap, isl_dim_in, pos, 1);
+
+  // get string back from ISL map
+  char * cstr;
+  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
+  isl_printer_print_map(ip ,imap);
+  cstr=isl_printer_get_str(ip);
+  string stringFromISL = cstr;
+  
+  // clean-up
+  isl_printer_flush(ip);
+  isl_printer_free(ip);
+  free(cstr);
+  isl_map_free(imap);
+  imap= NULL;
+  isl_ctx_free(ctx); 
+
+  return stringFromISL;
+}
+
+std::string set_project_out (std::string relstr, unsigned pos)
```

Comment:
The above and this one should be encapsulated in SparseConstraints::project_out.